pages/create/index.page.js

```diff
@@ -1,35 +1,78 @@
-import Button from "../../components/button/button";
-import Input from "../../components/input/input";
-import Select from "../../components/select/select";
-import { Container, Form, Title } from "./create.styles";
 import { useForm } from "react-hook-form";
 import { number, object, string, date, ref } from "yup";
 import { yupResolver } from "@hookform/resolvers/yup";
+import { useContractWrite, usePrepareContractWrite } from "wagmi";
+import useDebounce from "../../hooks/useDebounce";
+
+import Select from "../../components/select/select";
+import crowdfundingConfig from "../../crowdfunding.config.json";
+
+import { Container, Form, Title } from "./create.styles";
+
+const validationSchema = object().shape({
+  title: string().required(),
+  subtitle: string().required(),
+  story: string().required(),
+  token: string().required(),
+  fundGoal: number().required(),
+  startDate: date()
+    .required()
+    .min(new Date(), "Date should be greater than now"),
+  endDate: date()
+    .required()
+    .min(ref("startDate"), "End date should be after start date"),
+});
 
 const Create = () => {
-  const validationSchema = object().shape({
-    title: string().required(),
-    subtitle: string().required(),
-    story: string().required(),
-    token: string().required(),
-    fundGoal: number().required(),
-    startDate: date()
-      .required()
-      .max(new Date(), "Created date can not be future"),
-    endDate: date()
-      .required()
-      .min(ref("startDate"), "End date should be after start date"),
-  });
+  const { abi, address } = crowdfundingConfig;
 
   const formOptions = {
     resolver: yupResolver(validationSchema),
     defaultValues: {},
     mode: "onChange",
   };
-  const { handleSubmit, reset, register, formState } = useForm(formOptions);
-  const { errors } = formState;
+  const {
+    handleSubmit,
+    reset,
+    register,
+    formState: { errors },
+    getValues,
+    setValue,
+  } = useForm(formOptions);
+  const debouncedStartDate = useDebounce(
+    new Date(getValues("startDate")).getTime(),
+    500
+  );
+  const debouncedEndDate = useDebounce(
+    new Date(getValues("endDate")).getTime(),
+    500
+  );
+  const debouncedFundGoal = useDebounce(getValues("fundGoal"), 500);
+
+  const isEnabled =
+    !!debouncedFundGoal && !!debouncedStartDate && !!debouncedEndDate;
+
+  const { config } = usePrepareContractWrite({
+    address,
+    abi,
+    functionName: "launch",
+    args: [
+      debouncedFundGoal,
+      debouncedStartDate?.toString(),
+      debouncedEndDate?.toString(),
+    ],
+    enabled: isEnabled,
+  });
+
+  const { write } = useContractWrite({
+    ...config,
+    onSuccess() {
+      console.log("launch");
```

Comment:
What do you think about adding a toast or something to give feedback to the user that the tx succeeded?

Reply:
Agree